usr/src/cmd/zdb/zdb_il.c

```diff
@@ -249,41 +261,43 @@ zil_prt_rec_setattr(zilog_t *zilog, int txtype, lr_setattr_t *lr)
 
 /* ARGSUSED */
 static void
-zil_prt_rec_acl(zilog_t *zilog, int txtype, lr_acl_t *lr)
+zil_prt_rec_acl(zilog_t *zilog, int txtype, void *arg)
 {
+	lr_acl_t *lr = (lr_acl_t*) arg;
+
 	(void) printf("%sfoid %llu, aclcnt %llu\n", prefix,
 	    (u_longlong_t)lr->lr_foid, (u_longlong_t)lr->lr_aclcnt);
 }
 
-typedef void (*zil_prt_rec_func_t)();
+typedef void (*zil_prt_rec_func_t)(zilog_t*, int, void*);
 typedef struct zil_rec_info {
 	zil_prt_rec_func_t	zri_print;
-	char			*zri_name;
+	const char		*zri_name;
 	uint64_t		zri_count;
 } zil_rec_info_t;
 
 static zil_rec_info_t zil_rec_info[TX_MAX_TYPE] = {
-	{	NULL,			"Total              " },
-	{	zil_prt_rec_create,	"TX_CREATE          " },
-	{	zil_prt_rec_create,	"TX_MKDIR           " },
-	{	zil_prt_rec_create,	"TX_MKXATTR         " },
-	{	zil_prt_rec_create,	"TX_SYMLINK         " },
-	{	zil_prt_rec_remove,	"TX_REMOVE          " },
-	{	zil_prt_rec_remove,	"TX_RMDIR           " },
-	{	zil_prt_rec_link,	"TX_LINK            " },
-	{	zil_prt_rec_rename,	"TX_RENAME          " },
-	{	zil_prt_rec_write,	"TX_WRITE           " },
-	{	zil_prt_rec_truncate,	"TX_TRUNCATE        " },
-	{	zil_prt_rec_setattr,	"TX_SETATTR         " },
-	{	zil_prt_rec_acl,	"TX_ACL_V0          " },
-	{	zil_prt_rec_acl,	"TX_ACL_ACL         " },
-	{	zil_prt_rec_create,	"TX_CREATE_ACL      " },
-	{	zil_prt_rec_create,	"TX_CREATE_ATTR     " },
-	{	zil_prt_rec_create,	"TX_CREATE_ACL_ATTR " },
-	{	zil_prt_rec_create,	"TX_MKDIR_ACL       " },
-	{	zil_prt_rec_create,	"TX_MKDIR_ATTR      " },
-	{	zil_prt_rec_create,	"TX_MKDIR_ACL_ATTR  " },
-	{	zil_prt_rec_write,	"TX_WRITE2          " },
+	{	NULL,			"Total              ", 0 },
+	{	zil_prt_rec_create,	"TX_CREATE          ", 0 },
```

Comment:
maybe changing this to the new-style `.zri_print=zil_prt_rec_create, .zri_name="TX_CREATE"` would be better (assuming the compiler is OK with leaving out `zri_count`)